mpas_analysis/ocean/ocean_modelvsobs.py

```diff
@@ -94,126 +97,39 @@ def ocn_modelvsobs(config, field):
-            # set the coordinates now that the dimensions have the same names
-            dsObs.coords['lat'] = dsObs['latCoord']
-            dsObs.coords['lon'] = dsObs['lonCoord']
-            dsObs.coords['Time'] = dsObs['calmonth']
-            dsObs.coords['month'] = ('Time', np.array(dsObs['calmonth'], int))
-            # no meaningful year since this is already a climatology
-            dsObs.coords['year'] = ('Time', np.ones(dsObs.dims['Time'], int))
-
-            dsObs = mpas_xarray.subset_variables(dsObs, [obsFieldName,
-                                                         'month'])
-            # Reorder dataset for consistence with other obs. data sets
-            dsObs = dsObs.transpose('Time', 'lat', 'lon')
-
-        # Set appropriate MLD figure labels
-        observationTitleLabel = \
-            "Observations (HolteTalley density threshold MLD)"
-        outFileLabel = "mldHolteTalleyARGO"
-        unitsLabel = 'm'
-
-    elif field == 'sst':
-
-        iselvals = {'nVertLevels': 0}
-
-        climStartYear = config.getint('oceanObservations',
-                                      'sstClimatologyStartYear')
-        climEndYear = config.getint('oceanObservations',
-                                    'sstClimatologyEndYear')
-        timeStart = datetime.datetime(year=climStartYear, month=1, day=1)
-        timeEnd = datetime.datetime(year=climEndYear, month=12, day=31)
-
-        if climStartYear < 1925:
-            period = 'pre-industrial'
-        else:
-            period = 'present-day'
-
-        if buildObsClimatologies:
-            dsObs = xr.open_mfdataset(obsFileName)
-            dsObs.rename({'time': 'Time'}, inplace=True)
-            dsObs = dsObs.transpose('Time', 'lat', 'lon')
-            dsObs = dsObs.sel(Time=slice(timeStart, timeEnd))
-            dsObs.coords['month'] = dsObs['Time.month']
-            dsObs.coords['year'] = dsObs['Time.year']
-
-        obsFieldName = 'SST'
+    setupFuncs = {'mld': _setup_mld,
+                  'sst': _setup_sst,
+                  'sss': _setup_sss}
+    setup = setupFuncs[field]
 
-        # Set appropriate figure labels for SST
-        observationTitleLabel = \
-            "Observations (Hadley/OI, {} {:04d}-{:04d})".format(period,
-                                                                climStartYear,
-                                                                climEndYear)
-        outFileLabel = "sstHADOI"
-        unitsLabel = r'$^o$C'
+    setupObsFuncs = {'mld': _setup_mld_obs,
+                     'sst': _setup_sst_obs,
+                     'sss': _setup_sss_obs}
+    setupObs = setupObsFuncs[field]
 
-    elif field == 'sss':
+    iselvals, fieldInTitle, timeSeriesSectionName, obsFileName, \
+        obsFieldName, observationTitleLabel,  outFileLabel, unitsLabel, \
+        obsDescriptor, comparisonDescriptor = setup(config,
+                                                    observationsDirectory)
```

Comment:
Perform some setup specific to the field being analyzed.  Again, this will likely be cleaner in #128.